Tasks/Common/sonarqube-common/sonarqube-common.ts

```diff
@@ -87,6 +90,28 @@ export function getSonarQubeEndpointFromInput(inputFieldName): SonarQubeEndpoint
     return new SonarQubeEndpoint(hostUrl, hostUsername, hostPassword);
 }
 
+// Returns, as an object, the contents of the 'report-task.txt' file created by SonarQube plugins
+// The returned object contains the following properties:
+//   projectKey, serverUrl, dashboardUrl, ceTaskId, ceTaskUrl
+export function getSonarQubeTaskReport(sonarPluginFolder: string) {
+    var reportFilePath:string = path.join(sonarPluginFolder, 'report-task.txt');
```

Comment:
This function does too much - it computes the location of the file and then reads it. I recommend you let the responsability of computing the path to a different function